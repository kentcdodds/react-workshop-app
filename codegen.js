module.exports = require('./dist/codegen').getCode
