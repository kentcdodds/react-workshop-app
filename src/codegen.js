// this generates the code to use in the entry file

const loadFiles = require('./load-files')

function getCode({cwd = process.cwd(), ignore, renderOptions} = {}) {
  const filesInfo = loadFiles({cwd, ignore})

  const imports = filesInfo.map(({id, fullFilePath, ext}) => {
    let loaders = ''
    if (ext === '.html') {
      loaders = '!raw-loader!'
    } else if (ext === '.md' || ext === '.mdx') {
      loaders = '!babel-loader!mdx-loader!'
    }
    return `"${id}": () => import("${loaders}${fullFilePath}")`
  })

  return `
import makeWorkshopApp from '@kentcdodds/react-workshop-app'
import pkg from '../package.json'

const filesInfo = ${JSON.stringify(filesInfo, null, 2)}

makeWorkshopApp({
  imports: {
    ${imports.join(',\n    ')}
  },
  filesInfo,
  projectTitle: pkg.title,
  ${renderOptions ? `renderOptions: ${JSON.stringify(renderOptions)}` : ''}
})`.trim()
}

module.exports = getCode
