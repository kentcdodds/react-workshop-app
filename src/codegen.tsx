// this generates the code to use in the entry file
import {getAppInfo} from './get-app-info'

function getCode({
  cwd = process.cwd(),
  ignore,
  options,
}: {
  cwd?: string
  ignore?: Array<string>
  options?: Record<string, unknown>
} = {}) {
  const {gitHubRepoUrl, filesInfo, hasBackend, imports} = getAppInfo({
    cwd,
    ignore,
  })
  return `
import {makeKCDWorkshopApp} from '@kentcdodds/react-workshop-app'
import {loadDevTools} from '@kentcdodds/react-workshop-app/dev-tools'
import pkg from '../package.json'
${hasBackend ? `import * as backend from './backend'` : ''}

if (module.hot) module.hot.accept()

const filesInfo = ${JSON.stringify(filesInfo, null, 2)}

loadDevTools(() => {
  makeKCDWorkshopApp({
    imports: {
      ${imports.join(',\n      ')}
    },
    filesInfo,
    projectTitle: pkg.title,
    gitHubRepoUrl: \`${gitHubRepoUrl}\`,
    ${hasBackend ? `backend,` : ''}
    ${options ? `options: ${JSON.stringify(options)},` : ''}
  })
})`.trim()
}

export {getCode}
