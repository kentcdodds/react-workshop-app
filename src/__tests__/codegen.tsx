import path from 'path'
import {getCode} from '../codegen'

const cwd = path
  .join(__dirname, '../../example/react-fundamentals')
  .split(path.sep)
  .join('/')

expect.addSnapshotSerializer({
  test(val: unknown): val is string {
    return typeof val === 'string'
  },
  print(val: unknown) {
    if (process.platform === 'win32') {
      val = (val as string).replace(/\\\\/g, '/')
    }
    return (val as string).split(cwd).join('<PROJECT_ROOT>')
  },
})

test('review the snapshot please', () => {
  expect(getCode({cwd})).toMatchSnapshot()
})
