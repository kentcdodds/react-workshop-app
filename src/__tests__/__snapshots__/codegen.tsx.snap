// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`review the snapshot please 1`] = `
import {makeKCDWorkshopApp} from '@kentcdodds/react-workshop-app'
import {loadDevTools} from '@kentcdodds/react-workshop-app/dev-tools'
import pkg from '../package.json'
import * as backend from './backend'

if (module.hot) module.hot.accept()

const filesInfo = [
  {
    "id": "src/examples/example.js",
    "title": "Unknown",
    "fullFilePath": "<PROJECT_ROOT>/src/examples/example.js",
    "filePath": "src/examples/example.js",
    "isolatedPath": "/isolated/examples/example.js",
    "ext": ".js",
    "filename": "example",
    "type": "examples",
    "number": 0,
    "isExtraCredit": false,
    "extraCreditNumber": 0,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/examples/non-exported-example.js",
    "title": "Unknown",
    "fullFilePath": "<PROJECT_ROOT>/src/examples/non-exported-example.js",
    "filePath": "src/examples/non-exported-example.js",
    "isolatedPath": "/isolated/examples/non-exported-example.js",
    "ext": ".js",
    "filename": "non-exported-example",
    "type": "examples",
    "number": 0,
    "isExtraCredit": false,
    "extraCreditNumber": 0,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/examples/ts-example.tsx",
    "title": "Unknown",
    "fullFilePath": "<PROJECT_ROOT>/src/examples/ts-example.tsx",
    "filePath": "src/examples/ts-example.tsx",
    "isolatedPath": "/isolated/examples/ts-example.tsx",
    "ext": ".tsx",
    "filename": "ts-example",
    "type": "examples",
    "number": 0,
    "isExtraCredit": false,
    "extraCreditNumber": 0,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/01.html",
    "title": "Basic JavaScript-rendered Hello World",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/01.html",
    "filePath": "src/exercise/01.html",
    "isolatedPath": "/isolated/exercise/01.html",
    "ext": ".html",
    "filename": "01",
    "type": "exercise",
    "number": 1,
    "isExtraCredit": false,
    "extraCreditNumber": 1,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/01.md",
    "title": "Basic JavaScript-rendered Hello World",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/01.md",
    "filePath": "src/exercise/01.md",
    "isolatedPath": "/isolated/exercise/01.md",
    "ext": ".md",
    "filename": "01",
    "type": "instruction",
    "number": 1,
    "isExtraCredit": false,
    "extraCreditNumber": 1,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/02.html",
    "title": "Intro to raw React APIs",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/02.html",
    "filePath": "src/exercise/02.html",
    "isolatedPath": "/isolated/exercise/02.html",
    "ext": ".html",
    "filename": "02",
    "type": "exercise",
    "number": 2,
    "isExtraCredit": false,
    "extraCreditNumber": 2,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/02.md",
    "title": "Intro to raw React APIs",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/02.md",
    "filePath": "src/exercise/02.md",
    "isolatedPath": "/isolated/exercise/02.md",
    "ext": ".md",
    "filename": "02",
    "type": "instruction",
    "number": 2,
    "isExtraCredit": false,
    "extraCreditNumber": 2,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/03.html",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/03.html",
    "filePath": "src/exercise/03.html",
    "isolatedPath": "/isolated/exercise/03.html",
    "ext": ".html",
    "filename": "03",
    "type": "exercise",
    "number": 3,
    "isExtraCredit": false,
    "extraCreditNumber": 3,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/03.md",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/03.md",
    "filePath": "src/exercise/03.md",
    "isolatedPath": "/isolated/exercise/03.md",
    "ext": ".md",
    "filename": "03",
    "type": "instruction",
    "number": 3,
    "isExtraCredit": false,
    "extraCreditNumber": 3,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/04.html",
    "title": "Creating custom components",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/04.html",
    "filePath": "src/exercise/04.html",
    "isolatedPath": "/isolated/exercise/04.html",
    "ext": ".html",
    "filename": "04",
    "type": "exercise",
    "number": 4,
    "isExtraCredit": false,
    "extraCreditNumber": 4,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/04.md",
    "title": "Creating custom components",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/04.md",
    "filePath": "src/exercise/04.md",
    "isolatedPath": "/isolated/exercise/04.md",
    "ext": ".md",
    "filename": "04",
    "type": "instruction",
    "number": 4,
    "isExtraCredit": false,
    "extraCreditNumber": 4,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/05.md",
    "title": "Styling",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/05.md",
    "filePath": "src/exercise/05.md",
    "isolatedPath": "/isolated/exercise/05.md",
    "ext": ".md",
    "filename": "05",
    "type": "instruction",
    "number": 5,
    "isExtraCredit": false,
    "extraCreditNumber": 5,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/05.tsx",
    "title": "Styling",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/05.tsx",
    "filePath": "src/exercise/05.tsx",
    "isolatedPath": "/isolated/exercise/05.tsx",
    "ext": ".tsx",
    "filename": "05",
    "type": "exercise",
    "number": 5,
    "isExtraCredit": false,
    "extraCreditNumber": 5,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/06.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06.js",
    "filePath": "src/exercise/06.js",
    "isolatedPath": "/isolated/exercise/06.js",
    "ext": ".js",
    "filename": "06",
    "type": "exercise",
    "number": 6,
    "isExtraCredit": false,
    "extraCreditNumber": 6,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/06.md",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06.md",
    "filePath": "src/exercise/06.md",
    "isolatedPath": "/isolated/exercise/06.md",
    "ext": ".md",
    "filename": "06",
    "type": "instruction",
    "number": 6,
    "isExtraCredit": false,
    "extraCreditNumber": 6,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/06 copy.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06 copy.js",
    "filePath": "src/exercise/06 copy.js",
    "isolatedPath": "/isolated/exercise/06 copy.js",
    "ext": ".js",
    "filename": "06 copy",
    "type": "exercise",
    "number": 6,
    "isExtraCredit": false,
    "extraCreditNumber": 0,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/06.extra-2.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06.extra-2.js",
    "filePath": "src/exercise/06.extra-2.js",
    "isolatedPath": "/isolated/exercise/06.extra-2.js",
    "ext": ".js",
    "filename": "06.extra-2",
    "type": "exercise",
    "number": 6,
    "isExtraCredit": true,
    "extraCreditNumber": 2,
    "extraCreditTitle": "another example extra (exercise)"
  },
  {
    "id": "src/exercise/07.js",
    "title": "Dynamic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/07.js",
    "filePath": "src/exercise/07.js",
    "isolatedPath": "/isolated/exercise/07.js",
    "ext": ".js",
    "filename": "07",
    "type": "exercise",
    "number": 7,
    "isExtraCredit": false,
    "extraCreditNumber": 7,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/07.md",
    "title": "Dynamic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/07.md",
    "filePath": "src/exercise/07.md",
    "isolatedPath": "/isolated/exercise/07.md",
    "ext": ".md",
    "filename": "07",
    "type": "instruction",
    "number": 7,
    "isExtraCredit": false,
    "extraCreditNumber": 7,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/08.js",
    "title": "Controlled Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/08.js",
    "filePath": "src/exercise/08.js",
    "isolatedPath": "/isolated/exercise/08.js",
    "ext": ".js",
    "filename": "08",
    "type": "exercise",
    "number": 8,
    "isExtraCredit": false,
    "extraCreditNumber": 8,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/08.md",
    "title": "Controlled Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/08.md",
    "filePath": "src/exercise/08.md",
    "isolatedPath": "/isolated/exercise/08.md",
    "ext": ".md",
    "filename": "08",
    "type": "instruction",
    "number": 8,
    "isExtraCredit": false,
    "extraCreditNumber": 8,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/exercise/09.js",
    "title": "Rendering Lists",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/09.js",
    "filePath": "src/exercise/09.js",
    "isolatedPath": "/isolated/exercise/09.js",
    "ext": ".js",
    "filename": "09",
    "type": "exercise",
    "number": 9,
    "isExtraCredit": false,
    "extraCreditNumber": 9,
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/09.md",
    "title": "Rendering Arrays",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/09.md",
    "filePath": "src/exercise/09.md",
    "isolatedPath": "/isolated/exercise/09.md",
    "ext": ".md",
    "filename": "09",
    "type": "instruction",
    "number": 9,
    "isExtraCredit": false,
    "extraCreditNumber": 9,
    "extraCreditTitle": "Unknown"
  },
  {
    "id": "src/final/01.html",
    "title": "Basic JavaScript-rendered Hello World",
    "fullFilePath": "<PROJECT_ROOT>/src/final/01.html",
    "filePath": "src/final/01.html",
    "isolatedPath": "/isolated/final/01.html",
    "ext": ".html",
    "filename": "01",
    "type": "final",
    "number": 1,
    "isExtraCredit": false,
    "extraCreditNumber": 1,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/01.extra-1.html",
    "title": "Basic JavaScript-rendered Hello World",
    "fullFilePath": "<PROJECT_ROOT>/src/final/01.extra-1.html",
    "filePath": "src/final/01.extra-1.html",
    "isolatedPath": "/isolated/final/01.extra-1.html",
    "ext": ".html",
    "filename": "01.extra-1",
    "type": "final",
    "number": 1,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "generate the root node"
  },
  {
    "id": "src/final/02.html",
    "title": "Intro to raw React APIs",
    "fullFilePath": "<PROJECT_ROOT>/src/final/02.html",
    "filePath": "src/final/02.html",
    "isolatedPath": "/isolated/final/02.html",
    "ext": ".html",
    "filename": "02",
    "type": "final",
    "number": 2,
    "isExtraCredit": false,
    "extraCreditNumber": 2,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/02.extra-1.html",
    "title": "Intro to raw React APIs",
    "fullFilePath": "<PROJECT_ROOT>/src/final/02.extra-1.html",
    "filePath": "src/final/02.extra-1.html",
    "isolatedPath": "/isolated/final/02.extra-1.html",
    "ext": ".html",
    "filename": "02.extra-1",
    "type": "final",
    "number": 2,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "nesting elements"
  },
  {
    "id": "src/final/03.html",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.html",
    "filePath": "src/final/03.html",
    "isolatedPath": "/isolated/final/03.html",
    "ext": ".html",
    "filename": "03",
    "type": "final",
    "number": 3,
    "isExtraCredit": false,
    "extraCreditNumber": 3,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/03.extra-1.html",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.extra-1.html",
    "filePath": "src/final/03.extra-1.html",
    "isolatedPath": "/isolated/final/03.extra-1.html",
    "ext": ".html",
    "filename": "03.extra-1",
    "type": "final",
    "number": 3,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "interpolate className and children"
  },
  {
    "id": "src/final/03.extra-2.html",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.extra-2.html",
    "filePath": "src/final/03.extra-2.html",
    "isolatedPath": "/isolated/final/03.extra-2.html",
    "ext": ".html",
    "filename": "03.extra-2",
    "type": "final",
    "number": 3,
    "isExtraCredit": true,
    "extraCreditNumber": 2,
    "extraCreditTitle": "spread props"
  },
  {
    "id": "src/final/03.extra-3.html",
    "title": "Using JSX",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.extra-3.html",
    "filePath": "src/final/03.extra-3.html",
    "isolatedPath": "/isolated/final/03.extra-3.html",
    "ext": ".html",
    "filename": "03.extra-3",
    "type": "final",
    "number": 3,
    "isExtraCredit": true,
    "extraCreditNumber": 3,
    "extraCreditTitle": "really long extra credit title. Like really long."
  },
  {
    "id": "src/final/04.html",
    "title": "Creating custom components",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.html",
    "filePath": "src/final/04.html",
    "isolatedPath": "/isolated/final/04.html",
    "ext": ".html",
    "filename": "04",
    "type": "final",
    "number": 4,
    "isExtraCredit": false,
    "extraCreditNumber": 4,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/04.extra-1.html",
    "title": "Creating custom components",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.extra-1.html",
    "filePath": "src/final/04.extra-1.html",
    "isolatedPath": "/isolated/final/04.extra-1.html",
    "ext": ".html",
    "filename": "04.extra-1",
    "type": "final",
    "number": 4,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "using a custom component with React.createElement"
  },
  {
    "id": "src/final/04.extra-2.html",
    "title": "Creating custom components",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.extra-2.html",
    "filePath": "src/final/04.extra-2.html",
    "isolatedPath": "/isolated/final/04.extra-2.html",
    "ext": ".html",
    "filename": "04.extra-2",
    "type": "final",
    "number": 4,
    "isExtraCredit": true,
    "extraCreditNumber": 2,
    "extraCreditTitle": "using a custom component with JSX"
  },
  {
    "id": "src/final/05.tsx",
    "title": "Styling",
    "fullFilePath": "<PROJECT_ROOT>/src/final/05.tsx",
    "filePath": "src/final/05.tsx",
    "isolatedPath": "/isolated/final/05.tsx",
    "ext": ".tsx",
    "filename": "05",
    "type": "final",
    "number": 5,
    "isExtraCredit": false,
    "extraCreditNumber": 5,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/05.extra-1.tsx",
    "title": "Styling",
    "fullFilePath": "<PROJECT_ROOT>/src/final/05.extra-1.tsx",
    "filePath": "src/final/05.extra-1.tsx",
    "isolatedPath": "/isolated/final/05.extra-1.tsx",
    "ext": ".tsx",
    "filename": "05.extra-1",
    "type": "final",
    "number": 5,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "Custom component that encapsulates styling"
  },
  {
    "id": "src/final/06.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/final/06.js",
    "filePath": "src/final/06.js",
    "isolatedPath": "/isolated/final/06.js",
    "ext": ".js",
    "filename": "06",
    "type": "final",
    "number": 6,
    "isExtraCredit": false,
    "extraCreditNumber": 6,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/06.extra-1.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/final/06.extra-1.js",
    "filePath": "src/final/06.extra-1.js",
    "isolatedPath": "/isolated/final/06.extra-1.js",
    "ext": ".js",
    "filename": "06.extra-1",
    "type": "final",
    "number": 6,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "using refs"
  },
  {
    "id": "src/final/06.extra-2.js",
    "title": "Basic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/final/06.extra-2.js",
    "filePath": "src/final/06.extra-2.js",
    "isolatedPath": "/isolated/final/06.extra-2.js",
    "ext": ".js",
    "filename": "06.extra-2",
    "type": "final",
    "number": 6,
    "isExtraCredit": true,
    "extraCreditNumber": 2,
    "extraCreditTitle": "another example extra (final)"
  },
  {
    "id": "src/final/07.js",
    "title": "Dynamic Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/final/07.js",
    "filePath": "src/final/07.js",
    "isolatedPath": "/isolated/final/07.js",
    "ext": ".js",
    "filename": "07",
    "type": "final",
    "number": 7,
    "isExtraCredit": false,
    "extraCreditNumber": 7,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/08.js",
    "title": "Controlled Forms",
    "fullFilePath": "<PROJECT_ROOT>/src/final/08.js",
    "filePath": "src/final/08.js",
    "isolatedPath": "/isolated/final/08.js",
    "ext": ".js",
    "filename": "08",
    "type": "final",
    "number": 8,
    "isExtraCredit": false,
    "extraCreditNumber": 8,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/09.js",
    "title": "Rendering Lists",
    "fullFilePath": "<PROJECT_ROOT>/src/final/09.js",
    "filePath": "src/final/09.js",
    "isolatedPath": "/isolated/final/09.js",
    "ext": ".js",
    "filename": "09",
    "type": "final",
    "number": 9,
    "isExtraCredit": false,
    "extraCreditNumber": 9,
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/09.extra-1.js",
    "title": "Rendering Lists",
    "fullFilePath": "<PROJECT_ROOT>/src/final/09.extra-1.js",
    "filePath": "src/final/09.extra-1.js",
    "isolatedPath": "/isolated/final/09.extra-1.js",
    "ext": ".js",
    "filename": "09.extra-1",
    "type": "final",
    "number": 9,
    "isExtraCredit": true,
    "extraCreditNumber": 1,
    "extraCreditTitle": "Focus Demo"
  },
  {
    "id": "src/index.js",
    "title": "Unknown",
    "fullFilePath": "<PROJECT_ROOT>/src/index.js",
    "filePath": "src/index.js",
    "isolatedPath": "/isolated/index.js",
    "ext": ".js",
    "filename": "index",
    "type": "src",
    "number": 0,
    "isExtraCredit": false,
    "extraCreditNumber": 0,
    "extraCreditTitle": "Unknown"
  }
]

loadDevTools(() => {
  makeKCDWorkshopApp({
    imports: {
      "src/examples/example.js": () => import("./examples/example.js"),
      "src/examples/non-exported-example.js": () => import("./examples/non-exported-example.js"),
      "src/examples/ts-example.tsx": () => import("./examples/ts-example.tsx"),
      "src/exercise/01.html": () => import("!raw-loader!./exercise/01.html"),
      "src/exercise/01.md": () => import("!babel-loader!mdx-loader!./exercise/01.md"),
      "src/exercise/02.html": () => import("!raw-loader!./exercise/02.html"),
      "src/exercise/02.md": () => import("!babel-loader!mdx-loader!./exercise/02.md"),
      "src/exercise/03.html": () => import("!raw-loader!./exercise/03.html"),
      "src/exercise/03.md": () => import("!babel-loader!mdx-loader!./exercise/03.md"),
      "src/exercise/04.html": () => import("!raw-loader!./exercise/04.html"),
      "src/exercise/04.md": () => import("!babel-loader!mdx-loader!./exercise/04.md"),
      "src/exercise/05.md": () => import("!babel-loader!mdx-loader!./exercise/05.md"),
      "src/exercise/05.tsx": () => import("./exercise/05.tsx"),
      "src/exercise/06.js": () => import("./exercise/06.js"),
      "src/exercise/06.md": () => import("!babel-loader!mdx-loader!./exercise/06.md"),
      "src/exercise/06 copy.js": () => import("./exercise/06 copy.js"),
      "src/exercise/06.extra-2.js": () => import("./exercise/06.extra-2.js"),
      "src/exercise/07.js": () => import("./exercise/07.js"),
      "src/exercise/07.md": () => import("!babel-loader!mdx-loader!./exercise/07.md"),
      "src/exercise/08.js": () => import("./exercise/08.js"),
      "src/exercise/08.md": () => import("!babel-loader!mdx-loader!./exercise/08.md"),
      "src/exercise/09.js": () => import("./exercise/09.js"),
      "src/exercise/09.md": () => import("!babel-loader!mdx-loader!./exercise/09.md"),
      "src/final/01.html": () => import("!raw-loader!./final/01.html"),
      "src/final/01.extra-1.html": () => import("!raw-loader!./final/01.extra-1.html"),
      "src/final/02.html": () => import("!raw-loader!./final/02.html"),
      "src/final/02.extra-1.html": () => import("!raw-loader!./final/02.extra-1.html"),
      "src/final/03.html": () => import("!raw-loader!./final/03.html"),
      "src/final/03.extra-1.html": () => import("!raw-loader!./final/03.extra-1.html"),
      "src/final/03.extra-2.html": () => import("!raw-loader!./final/03.extra-2.html"),
      "src/final/03.extra-3.html": () => import("!raw-loader!./final/03.extra-3.html"),
      "src/final/04.html": () => import("!raw-loader!./final/04.html"),
      "src/final/04.extra-1.html": () => import("!raw-loader!./final/04.extra-1.html"),
      "src/final/04.extra-2.html": () => import("!raw-loader!./final/04.extra-2.html"),
      "src/final/05.tsx": () => import("./final/05.tsx"),
      "src/final/05.extra-1.tsx": () => import("./final/05.extra-1.tsx"),
      "src/final/06.js": () => import("./final/06.js"),
      "src/final/06.extra-1.js": () => import("./final/06.extra-1.js"),
      "src/final/06.extra-2.js": () => import("./final/06.extra-2.js"),
      "src/final/07.js": () => import("./final/07.js"),
      "src/final/08.js": () => import("./final/08.js"),
      "src/final/09.js": () => import("./final/09.js"),
      "src/final/09.extra-1.js": () => import("./final/09.extra-1.js"),
      "src/index.js": () => import("./index.js")
    },
    filesInfo,
    projectTitle: pkg.title,
    gitHubRepoUrl: \`https://github.com/kentcdodds/react-workshop-app\`,
    backend,
    
  })
})
`;
